java/plasma/src/main/java/org/apache/arrow/plasma/PlasmaClient.java

```diff
@@ -45,18 +46,9 @@ public PlasmaClient(String storeSocketName, String managerSocketName, int releas
   // interface methods --------------------
 
   @Override
-  public void put(byte[] objectId, byte[] value, byte[] metadata) {
-    ByteBuffer buf = null;
-    try {
-      buf = PlasmaClientJNI.create(conn, objectId, value.length, metadata);
-    } catch (Exception e) {
-      System.err.println("ObjectId " + objectId + " error at PlasmaClient put");
-      e.printStackTrace();
-    }
-    if (buf == null) {
-      return;
-    }
-
+  public void put(byte[] objectId, byte[] value, byte[] metadata)
+          throws DuplicateObjectException, PlasmaOutOfMemoryException {
+    ByteBuffer buf = PlasmaClientJNI.create(conn, objectId, value.length, metadata);
```

Comment:
this looks backward incompatible..earlier we were ignoring any exceptions while creating the buffer

which is why the test is failing here - https://github.com/apache/arrow/blob/41a4e72f08021a682426ceb46c7921ad91a269cd/java/plasma/src/test/java/org/apache/arrow/plasma/PlasmaClientTest.java#L145

any reason to change the behavior?

Reply:
The reason is because it makes more sense to propagate the exception to user code, and let user decide what to do. E.g., when putting an object with existing ID, someone may want to silently ignore the error, someone may want to use a different ID, and someone may want to terminate the program. The decision should depends on application logic.